File: AndroidFx/app/src/main/java/com/android/custview/ui/ChangeSkinActivity.java
package com.android.custview.ui;

import android.content.res.Configuration;
import android.view.View;

import androidx.appcompat.app.AppCompatDelegate;

import com.android.custview.R;
import com.android.zp.base.BaseActivity;
import com.android.zp.base.KLog;
import com.android.zp.base.utils.PreferencesUtils;

public class ChangeSkinActivity extends BaseActivity {

    @Override
    public int getLayout() {
        return R.layout.change_skin_layout;
    }

    @Override
    public void initView() {
    }

    @Override
    public void initData() {

    }

    // 点击事件
    public void dayOrNight(View view) {
        int uiMode = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
        KLog.logE("uiMode: " + uiMode);
        switch (uiMode) {
            case Configuration.UI_MODE_NIGHT_NO:
                setDayNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                PreferencesUtils.putBoolean(this, "isNight", true);
                break;
            case Configuration.UI_MODE_NIGHT_YES:
                setDayNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                PreferencesUtils.putBoolean(this, "isNight", false);
                break;
            default:
                break;
        }
    }

    @Override
    protected boolean openChangeSkin() {
        return true;
    }
}

File: AndroidFx/app/src/main/java/com/android/custview/ui/ChangeSkinActivity.kt
package com.android.custview.ui

import android.content.res.Configuration
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatDelegate
import com.android.custview.R
import com.android.zp.base.SkinActivity
import com.blankj.utilcode.util.SPUtils


class ChangeSkinActivity : SkinActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.change_skin_layout)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)

        val isNight = SPUtils.getInstance().getBoolean("isNight", false)
        if (isNight) {
            setDayNightMode(AppCompatDelegate.MODE_NIGHT_YES)
        } else {
            setDayNightMode(AppCompatDelegate.MODE_NIGHT_NO)
        }
    }

    override fun openChangeSkin(): Boolean {
        return true
    }

    fun dayOrNight(View: View) {
        when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
            Configuration.UI_MODE_NIGHT_NO -> {
                setDayNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                SPUtils.getInstance().put("isNight", false)
            }
            Configuration.UI_MODE_NIGHT_YES -> {
                setDayNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                SPUtils.getInstance().put("isNight", true)
            }
        }

    }
}